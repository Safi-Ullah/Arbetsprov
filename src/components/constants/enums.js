export const Enums = {
    TASK_STATUS: {
        created: 1,
        active: 2,
        completed: 3,
        onhold: 4,
        deleted: 5
    },
    TASK_ACTIONS: {
        edit: 1,
        setComplexity: 2,
        delete: 3,
    }
};