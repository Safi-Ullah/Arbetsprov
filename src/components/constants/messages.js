export const PromptMessages = {
    DELETE_TASK_PROMPT: `Are you sure you want to permanently delete this task?
        You will not be able to restore in the future.`
};