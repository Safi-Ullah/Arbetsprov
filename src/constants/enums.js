export const Enums = {
    TASK_STATUS: {
        created: 1,
        active: 2,
        completed: 3,
        onhold: 4,
        deleted: 5
    },
    TASK_ACTIONS: {
        edit: 1,
        setComplexity: 2,
        delete: 3,
    },
    APP_SCREENS: {
        dashboard: 1,
        deletedTasks: 2
    },
    SORT_BY_COMPLEXITY: {
        ascending: 1,
        descending: 2
    },
    TEST_TYPES: {
        dom: 'Dom test',
        snapshot: 'Snapshot test'
    }
};