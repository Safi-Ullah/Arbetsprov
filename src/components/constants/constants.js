import { Enums } from './enums';

export const TASK_ACTION_HANDLERS = {
    [Enums.TASK_ACTIONS.edit]: 'displayEditTaskModal',
    [Enums.TASK_ACTIONS.delete]: 'displayConfirmationDialog'
};