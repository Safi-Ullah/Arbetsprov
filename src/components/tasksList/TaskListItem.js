import React from 'react';
import PropTypes from 'prop-types';
import { Enums, TASK_ACTION_HANDLERS } from '../constants';

export const TaskListItem = props => {

    const { displayConfirmationDialog, task } = props;

    const taskActions = {
        'Edit': Enums.TASK_ACTIONS.edit,
        'Delete': Enums.TASK_ACTIONS.delete
    };

    return <li className="list-group-item">
        <span className="col-sm-8">{task.title}</span>
        {
            task.complexity
                ? <span className="badge badge-danger">{task.complexity}</span>
                : null
        }
        <button
            className="close ml-2"
            type="button"
            id="dropdownMenuButton"
            data-toggle="dropdown"
            aria-haspopup="true"
            aria-expanded="false">
            <span aria-hidden="true">&Xi;</span>
        </button>
        <div className="dropdown-menu" aria-labelledby="dropdownMenuButton">
            {
                Object.keys(taskActions).map(action => <button
                    key={taskActions[action]}
                    onClick={() => props[TASK_ACTION_HANDLERS[taskActions[action]]](task.id)}
                    className="dropdown-item">
                    { action }
                </button>)
            }
        </div>
        <button type="button" className="close"
            onClick={() => displayConfirmationDialog(task.id)}>
            <span aria-hidden="true" className="delete-btn">&times;</span>
        </button>
    </li>
};

TaskListItem.propTypes = {
    displayConfirmationDialog: PropTypes.func.isRequired,
    task: PropTypes.object.isRequired
};