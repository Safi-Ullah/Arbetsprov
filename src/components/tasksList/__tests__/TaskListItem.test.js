import React from 'react';
import TestRenderer from 'react-test-renderer';
import { TaskListItem } from '../TaskListItem';
import { Enums } from '../../../constants';

const props = {
    task: {
        title: 'test',
        desc: 'test',
        complexity: '0',
        assignee: 'test',
        status: Enums.TASK_STATUS.created
    },
    displayConfirmationDialog: jest.fn(),
    displayEditTaskModal: jest.fn(),
    taskNextStatus: [],
    changeStatus: jest.fn()
};

describe(Enums.TEST_TYPES.snapshot, () => {
    it('show taskslist item', () => {
        const tree = TestRenderer.create(<TaskListItem {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});