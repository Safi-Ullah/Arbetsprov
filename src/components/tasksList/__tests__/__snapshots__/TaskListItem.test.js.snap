// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test show taskslist item 1`] = `
<li
  className="list-group-item"
>
  <button
    className="btn btn-link text-break"
    onClick={[Function]}
  >
    test
  </button>
  <span
    className="badge badge-info"
  >
    0
  </span>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    className="close ml-2"
    data-toggle="dropdown"
    id="dropdownMenuButton"
    type="button"
  >
    <span
      aria-hidden="true"
    >
      ⁝
    </span>
  </button>
  <div
    aria-labelledby="dropdownMenuButton"
    className="dropdown-menu"
  >
    <button
      className="dropdown-item"
      onClick={[Function]}
    >
      Edit
    </button>
  </div>
  <button
    className="close"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden="true"
      className="delete-btn"
    >
      ×
    </span>
  </button>
</li>
`;
